psychopy/tests/test_psychojs/test_py2js.py

```diff
@@ -0,0 +1,50 @@
+
+"""Test the output of py2js conversion
+"""
+
+import psychopy.experiment.py2js as py2js
+
+class Test_PY2JS_Compile(object):
+    """
+    Test class for py2js code conversion
+    """
+
+    def test_Py2js_Expression2js(self):
+        """Test that converts a short expression (e.g. a Component Parameter) Python to JS"""
+        input = ['sin(t)',
+                 'cos(t)',
+                 'tan(t)',
+                 'pi',
+                 'rand',
+                 'random',
+                 't*5',
+                 '(3, 4)',
+                 '(5*-2)',
+                 '(1,(2,3))',
+                 '2*(2, 3)',
+                 '[1, (2*2)]',
+                 '(.7, .7)',
+                 '(-.7, .7)',
+                 '[-.7, -.7]',
+                 '[-.7, (-.7 * 7)]']
+
+        output = ['Math.sin(t)',
+                  'Math.cos(t)',
+                  'Math.tan(t)',
+                  'Math.PI',
+                  'Math.random',
+                  'Math.random',
+                  '(t * 5)',
+                  '[3, 4]',
+                  '(5 * (- 2))',
+                  '[1, [2, 3]]',
+                  '[2 * [2, 3]]',
+                  '[1, (2 * 2)]',
+                  '[0.7, 0.7]',
+                  '[[- 0.7], 0.7]',
+                  '[[- 0.7], [- 0.7]]',
+                  '[(- 0.7), ((- 0.7) * 7)]']
+
+        for idx, expr in enumerate(input):
+            assert py2js.expression2js(expr) == output[idx]
```

Comment:
It looks like this fails because of a simple difference in number of spaces (output in Python2.7 is `'(5 * (-2))'` whereas in Py3.6 it's `'(5 * (- 2))'`. I'd suggest:
```
# check whether direct match or at least a match when spaces removed
assert (py2js.expression2js(expr) == output[idx] or 
        py2js.expression2js(expr).replace(" ", "") == output[idx].replace(" ", ""))
```

Reply:
Cheers. There is also a problem where ```ast.UnaryOp``` do not seem to be recognised in Python 2, so the decision of whether a node is a unary op is always false. Just trying to figure out why.